File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        max_num = max(nums)
        count = [0] * (max_num + 1)
        for i in range(len(nums)):
            count[nums[i]] += 1

        sorted_arr = []
        for i in range(len(count)):
            while count[i] > 0:
                sorted_arr.append(i)
                count[i] -= 1

        res = []
        for i in range(len(nums)):
            res.append(sorted_arr.index(nums[i]))

        return res

File: 1482-how-many-numbers-are-smaller-than-the-current-number/how-many-numbers-are-smaller-than-the-current-number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function (nums) {
    let max = Math.max(...nums)
    let count = Array(max + 1).fill(0)
    for (let i = 0; i < nums.length; i++) {
        count[nums[i]]++
    }
    let sortedArr = []
    for (let i = 0; i < count.length; i++) {
        while (count[i] > 0) {
            sortedArr.push(i)
            count[i]--
        }
    }
    let res = []
    let map = new Map()
    let cnt = 0
    for (let i = 0; i < sortedArr.length; i++) {
        if (map.get(sortedArr[i]) == undefined) {
            map.set(sortedArr[i], cnt)
        }
        cnt++
    }
    for (let i = 0; i < nums.length; i++) {
        res.push(map.get(nums[i]))
    }
    return res
};